### PR TITLE
feat: backend unit tests

### PR DESCRIPTION
**Pull request by SiyangShao**
# Description
This PR fixes https://github.com/ServerlessLLM/ServerlessLLM/issues/37 and add unit tests for backends
# Changes
+ Fix unit test for vLLM backends
+ Add unit test for transformer backends
+ Move ray definitions into runtime

### CODE REVIEW

**Comment by future-xy** [comment]
Could you try to add this to Gtihub workflow? Does it rely on GPU servers?

**Comment by SiyangShao** [comment]
I add this to Github Workflow. It do not rely on GPU servers since all GPU related operations are mocked.

File: tests/backend_test/vllm_backend_test.py [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+def backend_config():
+    return {
+        "pretrained_model_name_or_path": "test-model",
+        "trace_debug": False,
+        "load_format": "sharded_state",
+        "torch_dtype": None,
+    }
+
+
+async def generate_result(
```
Comment (future-xy) [inline tests/backend_test/vllm_backend_test.py:41]: Generate

File: tests/backend_test/vllm_backend_test.py [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+    with patch(
+        "serverless_llm.serve.backends.vllm_backend.AsyncLLMEngine.from_engine_args",
+        return_value=async_llm_engine,
+    ):
+        await vllm_backend.init_backend()
+    assert vllm_backend.status == BackendStatus.RUNNING
+
+
+@pytest.mark.asyncio
+async def test_generate_without_init(vllm_backend):
```
Comment (future-xy) [inline tests/backend_test/vllm_backend_test.py:110]: Should also add in Transformers backend

**Comment by SiyangShao** [comment]
Fixed backend test for transformer backend by change the load model using transformer's default load method and load the model to CPU only

**Review by future-xy — APPROVED** [review]

**Pull request merged**